### PR TITLE
[Enhancement] Fix #5042 Add copy link button

### LINKED ISSUE #5042: Add a direct "Copy Link" button next to Share button
### What is the user problem or growth opportunity you want to see solved?

Add a "Copy Link" button to the menu above articles to allow directly copying the link to the clipboard without going through Android's native Share menu.

### How do you know that this problem exists today? Why is this important?

This UX improvement makes sharing less effortful by reducing the number of taps (from 3 to 2) and preventing 1 to 3 seconds of delay on older phones to open Android's native Share menu.

### Who will benefit from it?

Many users, like me, who primarily need the link to the article to paste it in another app where in they're half way through writing a text or message.

### Anything else you would like to add?

Twitter has implemented in in it's Android app a long while ago. This improvement to the UX has made it easier and faster to copy links, taking only two (instead of three) taps clicking on Share button, and select:

Screenshot_20220901-094547-160

### PR DESCRIPTION
**Pull request by Justweng**
**Description**

Fixes #5042
Add a copy link button next to the share link button.

**Tests performed**

Tested MediaDetailPagerFragment on Pixel 4 with API level 28.

**Screenshots**

image

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

**Review by junjiemu62 — APPROVED** [review]

**Comment by whym** [comment]
Thanks for working on this! I tested it with Contributions and Explore and the new function worked well.

One thing I noticed is that the one additional button might not be ideal with a narrow screen. "Contributions" was not fully displayed when I tested with a 1080x1920 420 dpi emulator.
image

Could the new button be hidden in the dot menu in a narrow screen, perhaps? (There might be other solutions, but it was the first one I thought about.)

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -96,6 +96,8 @@
   <string name="menu_from_camera">Take photo</string>
   <string name="menu_nearby">Nearby</string>
   <string name="provider_contributions">My uploads</string>
+  <string name="menu_copy_link">Copy Link</string>
+  <string name="menu_link_copied">The link has copied to the clipboard</string>
```
Comment (whym) [inline app/src/main/res/values/strings.xml:100]: I think this should be "The link has been copied to the clipboard".

**Comment by shankarpriyank** [comment]
@nicolas-raoul This pr should be good to go after the changes in the strings?

**Comment by nicolas-raoul** [comment]
@Justweng Are you still working on this?
If not, someone may pull your pull request in their own branch to finish the work. 🙂

**Review by nicolas-raoul — APPROVED** [review]
Works great in both Contributions and Explore, thanks!

**Comment by nicolas-raoul** [comment]
> the one additional button might not be ideal with a narrow screen

This can be the topic of another issue. :-)

**Pull request merged**